app/public/js/profile.js

```diff
@@ -4,41 +4,18 @@ profile.js is used in the profile.blade.php view.
 
 var regions = [];
 
-/**
-When select-all checkboxes are checked, check every other associated checkbox.
-*/
-function selectAllChanged() {
-	if ( $(this).is(':checked') ) {
-		var tab = $(this).closest('[role="tabpanel"]');
-		var questions = tab.find('.questions input[type="checkbox"]');
-		questions.prop('checked', true);
-	}
-}
-
-function updateSelectAll() {
-	var tab = $(this).closest('[role="tabpanel"]');
-	var selectAll = tab.find('input.select-all');
-	var checkboxes = tab.find('.questions input[type="checkbox"]');
-	var allChecked = true;
-	checkboxes.each(function() {
-		if ( !$(this).is(':checked') )
-			allChecked = false;
-	});
-	selectAll.prop('checked', allChecked);
-}
-
-function initSelectAllBindings() {
-	var interests_box = $('.accessibility-interests');
-	var select_all_elements = interests_box.find('input.select-all');
-	var checkboxes = interests_box.find('.questions input');
-	var question_categories = interests_box.find('.category > .checkbox');
-	select_all_elements.change(selectAllChanged);
-	checkboxes.change(updateSelectAll);
-	question_categories.each(function() {
-		updateSelectAll.call(this);
-	});
+function selectAllToggle() {
+	$('button.select-all').click(function(){
+		if($(this).text() === "Select All") {
+		$(this).closest("div.category").find("input:checkbox").prop('checked',true);
+		$(this).text('Unselect All');
+		} else {
+		$(this).closest("div.category").find("input:checkbox").prop('checked',false);
+		//$(this).closest("div.category").find("input:checkbox").removeAttr('checked');
```

Comment:
I removed this commenting line. I have kept the removeAttr before, but it would not behave properly. Therefore, I add to change the prop to false. I refract that code because it is difficult to read. and the constructure is not clear.